doc/1.4/language.md

```diff
@@ -3395,116 +3395,129 @@ the implementations are provided by hierarchically unrelated templates such that
-    // side-effects, which is undesirable.
-    // Instead, very_complex_register is chosen as the base implementation
-    // called, and the behaviour of gated_register is replicated around that
-    // call.
-    method write_register(uint64 written, uint64 enabled_bytes,
-                          void *aux) default {
-        if (write_allowed()) {
-            this.templates.very_complex_register.write_register(
-                written, enabled_bytes, aux);
-        } else {
-            on_write_attempted_when_not_allowed();
-        }
-    }
+template write_1_clears is (write, get) { // unchanged
+    ...
 }
 
-in each (gated_register, very_complex_register) {
-    is very_complex_gated_register;
+template gated_write_1_clears is (gated_write, write_1_clears) {
+    // Resolve the conflict by providing an unambiguously most specific
+    // write implementation
+    method write(uint64 val) default {
+        // This makes gated_write the first link in the call chain...
+        this.templates.gated_write.write(val);
+    }
+
+    // ... and by overriding gated_write's base method, we make write_1_clears
+    // the second (and final) link in the chain
+    method base_write_of_gated_write(uint64 val) default {
+        this.templates.write_1_clears.write(val);
+    }
 }
+
+in each (gated_write, write_1_clears) { is gated_write_1_clears; }
 ```
+> [!NOTE]
+> The example given above applies regardless of whether the method
+> implementations in the original templates are `shared` or not. However, the
+> implementations in the template defined to resolve the conflicts might need to
+> be non-`shared` if some of the implementations involved are not `shared`;
+> see the final paragraph of this subsection.
+
+This approach extends to any number of conflicting templates &mdash; as long as
+all but (optionally) one can be made to offer overridable base methods (with
+distinct names.)
+
+Note that the order of implementations in the call chain might matter. If, in the
+example above, `write_1_clears` were instead modified to offer a base method
+and been made the first link in the chain, then what it would have passed down
+as the written value to `gated_write` would not have been the original value
+&mdash; but rather, the register's current value with bits cleared, which might
+violate the expectations of `gated_write`'s implementation.
+
+The approach given above is not the only way in which template-qualified method
+implementation calls may be utilized to resolve conflicts, only the most
+commonly applicable one. In fact, the most simple approach, if viable, is to
+have the implementation introduced to resolve the conflict simply call each
+conflicting implementation in turn &mdash; if that doesn't cause side-effects
+to be duplicated in an undesirable way &mdash; or even only call one particular
+implementation &mdash; if it makes sense to prefer it ahead of every other.
```

Comment:
The triple mdash in one sentence doesn't look good. Maybe split the sentence and replace them with commas?